hal/targets/hal/TARGET_STM/TARGET_STM32F4/common_objects.h

```diff
@@ -68,6 +68,21 @@ struct serial_s {
 #endif
 };
 
+struct i2c_s {
+    I2CName  i2c;
+    I2C_HandleTypeDef handle;
+    IRQn_Type event_i2cIRQ;
+    IRQn_Type error_i2cIRQ;
+    uint32_t slave;
+#if DEVICE_I2C_ASYNCH
+    uint32_t event;
+    uint32_t address;
```

Comment:
Indeed Martin, there is romm for improvment here.
Thought, "address" is given as uint32_t in I2C API, so it might safer to keep it as uint32_t in i2c_s struct. But we can fix some others.